src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/ProjectHotReloadSessionManager.cs

```diff
@@ -360,6 +369,12 @@ private async Task<bool> StopProjectAsync(HotReloadState hotReloadState, Cancell
                         }
                     }
                 }
+
+                // No more sessions removes the project from hot reload mode
+                if (_activeSessions.Count == 0)
+                {
+                    await _projectHotReloadNotificationService.Value.SetHotReloadStateAsync(isInHotReload: false);
```

Comment:
>SetHotReloadStateAsync [](http://example.com/codeflow?start=69&length=22)

the hotReloadState.Process.Kill call above may throw if the process exits between the HasExited call and the kill skipping this call. Does that matter? May want to consider putting this into a finally block or something isolated from that race.